spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/zuul/SampleZuulProxyApplicationTests.java

```diff
@@ -146,7 +146,7 @@ public void simpleHostRouteWithEncodedQuery() {
 				"http://localhost:" + this.port + "/self/query?foo={foo}", HttpMethod.GET,
 				new HttpEntity<>((Void) null), String.class, "weird#chars");
 		assertEquals(HttpStatus.OK, result.getStatusCode());
-		assertEquals("/query?foo=weird#chars", result.getBody());
+		assertEquals("/query?foo=weird%23chars", result.getBody());
```

Comment:
@spencergibb you're right the '#' is a gen-delims as specified at RFC, I didn't payed attention on that. 
Anyhow, the URI encoding is a real problem. I'll fix that PR.